attributes/default.rb

```diff
@@ -238,26 +241,55 @@
   'default' => "659400f9842fffb8d64ae0b650f081b9",
   'ubuntu' => { 'default' => "c2b410031867dc7c966ca5b1aa0c72aa" }
 )
+
+# MySQL Packages
+default['cluster']['mysql']['package']['root'] = "https://#{node['cluster']['region']}-aws-parallelcluster.s3.#{node['cluster']['region']}.#{node['cluster']['aws_domain']}/archives/mysql"
+default['cluster']['mysql']['package']['version'] = "8.0.31-1"
+default['cluster']['mysql']['package']['source-version'] = "8.0.31"
+default['cluster']['mysql']['package']['platform'] = if arm_instance?
+                                                       value_for_platform(
+                                                         'default' => "el/7/aarch64"
+                                                       )
+                                                     else
+                                                       value_for_platform(
+                                                         'default' => "el/7/x86_64",
+                                                         'ubuntu' => {
+                                                           '20.04' => "ubuntu/20.04/x86_64",
+                                                           '18.04' => "ubuntu/18.04/x86_64",
+                                                         }
+                                                       )
+                                                     end
+default['cluster']['mysql']['package']['file-name'] = "mysql-community-client-#{node['cluster']['mysql']['package']['version']}.tar.gz"
+default['cluster']['mysql']['package']['archive'] = "#{node['cluster']['mysql']['package']['root']}/#{node['cluster']['mysql']['package']['platform']}/#{node['cluster']['mysql']['package']['file-name']}"
+default['cluster']['mysql']['package']['source'] = "#{node['cluster']['mysql']['package']['root']}/source/mysql-#{node['cluster']['mysql']['package']['source-version']}.tar.gz"
+
+# MySQL Validation
 if arm_instance?
   default['cluster']['mysql']['repository']['packages'] = value_for_platform(
-    'default' => %w(mysql-community-devel mysql-community-libs mysql-community-common mysql-community-client-plugins),
+    'default' => %w(mysql-community-devel mysql-community-libs mysql-community-common mysql-community-client-plugins mysql-community-libs-compat),
     'ubuntu' => {
-      'default' => %w(libmysqlclient-dev libmysqlclient21 mysql-common),
-      '18.04' =>  %w(libmysqlclient-dev libmysqlclient20 mysql-common),
+      'default' => %w(libmysqlclient-dev libmysqlclient21),
+      '18.04' =>  %w(libmysqlclient-dev libmysqlclient20),
     }
   )
-  default['cluster']['mysql']['repository']['expected']['source'] = value_for_platform(
-    'default' => "mysql80-community",
-    'ubuntu' => { 'default' => "http://#{node['cluster']['region']}.ec2.ports.ubuntu.com/ubuntu-ports" }
+  default['cluster']['mysql']['repository']['expected']['version'] = value_for_platform(
+    'default' => "8.0.31",
```

Comment:
this is equal to `default['cluster']['mysql']['package']['source-version']`. Why do we need to redefine the version?

Reply:
The source version doesn't include the release, but I could redefine the package versions to be ['version'] + ['release'], what do you think?

Reply:
Thanks for the explanation. No, please keep the current approach.

Reply:
I would consider using a variable for this (and subsequent) version checks. This will definitely save us a build failure in the future.